library/src/main/java/com/novoda/downloadmanager/FileCallbackThrottleByProgressIncrease.java

```diff
@@ -14,7 +14,7 @@ public void setCallback(DownloadBatchStatusCallback callback) {
     @Override
     public void update(DownloadBatchStatus currentDownloadBatchStatus) {
         if (callback == null) {
-            return;
+            throw new IllegalStateException("A DownloadBatchStatusCallback must be set before an update is called.");
```

Comment:
I'm a little worried about introducing this change but it is a valid change that we should make.

Reply:
On one hand I agree, but on the other then I'll refactor how this callback works: 
in `stopUldates()` we set the callback to null, so we are not implying that we are not forwarding anything anymore (callback is null) but also that we won't receive any further update from downstream (because otherwise we'll crash).
This second assumption is wrong in my opinion...

Reply:
Sorry but I don't understand, it might be because of the number of negations in the paragraph 😂 You are saying you don't agree with this change? I'm not sure I follow the refactoring you propose.

Reply:
Sorry, what I meant is that I don't agree with this change if it's alone because we are connecting two external factors which in my opinion should stay disconnected, from a user of this class point of view:

- **output**: it is controlled by the method `stopUpdates()`, where we stop forwarding events to the output callback by setting its reference to `null`
- **input**: it is this method `update(DownloadBatchStatus currentDownloadBatchStatus)`, where we might continue receiving updates, but before this PR we were just doing nothing if the output was stopped, while now we crash

So the issue for me is that the user of this class needs to take care about its state. It made more sense to me before, having source (emitting updates) and controller (stopping output) possibly done by two different entities.

Reply:
Fair enough. How about I revert these two changes and replace with a warning log?

Reply:
Sounds good 👍